### ISSUE TITLE
Move protocol-common and protocol-v5 to separate repos

### ISSUE THREAD

**Issue by steamroller-airmash** [steamroller-airmash/airmash-server#35]
These are starting to approach stability, so it's worth spinning these out into their own repositories so that they can be depended on by other projects that want to use them.

In terms of use for the server, I plan on creating a bot framework to be used for testing and splitting these two packages off (and preferably putting them on [crates.io](https://crates.io)) would be useful for this.

**Comment by xyz-airmash** [comment]
What is the difference between common and v5?

**Comment by steamroller-airmash** [comment]
TLDR; common is an abstract protocol definition, v5 serializes stuff declared in common to the current airmash binary packet format.

protocol-common is the definitions that are meant to be used by clients. Its supposed to be a (somewhat) abstracted version of the airmash protocol that can be evolved as the server requires it. 

On the other hand, protocol-v5 is meant to serialize `ClientPacket` and `ServerPacket` declared in protocol-common to the wire protocol that the current version of the airmash protocol (v5) uses. Since it isn't really possible to change the wire protocol, protocol-v5 needs much stronger backwards compatibility in terms of outputs than what is required for definitions in protocol-common.

The reason for the split was so that new protocols can be introduced and use the machinery provided by protocol-common to be dropped in and work with the server as-is. It also allows protocol-v5 to (sort-of) polyfill in new packets that my be added in the future to protocol-common.

**Comment by steamroller-airmash** [comment]
For an example of polyfilling that would be useful today, right now a packet that uses a small array can only hold 255 items. Packets that use this include: EventRepel, PlayerHit, and others. It would be possible for protocol-v5 to transparently split this into multiple packets so that it doesn't have to be dealth with within server code.

**Comment by xyz-airmash** [comment]
Okay!
